src/sidebar/helpers/view-filter.js

```diff
@@ -35,6 +37,9 @@ function displayName(ann) {
 
 /**
  * Normalize a field value or query term for comparison.
+ *
+ * @template {string|number} T
+ * @param {T} val
```

Comment:
Can you help me understand why not `@param {string|number} T` here? i.e. why do we need a `@template`?